quipucords/scanner/network/utils.py

```diff
@@ -173,3 +182,58 @@ def get_fact_names():
 def raw_facts_template():
     """Results template for fact collection on network scans."""
     return {fact_name: None for fact_name in get_fact_names()}
+
+
+GEN_KEYFILES_DIRECTORY = "quipucords_generated_ssh_private_keys"
+GEN_KEYFILE_PREFIX = "quipucords_ssh_private_key_credential"
+
+
+def generate_ssh_keyfile(credential, ssh_keyvalue):
+    """Generate an ssh private keyfile for the ssh key content specified."""
+    ssh_pkey_dir = Path(tempfile.gettempdir()) / GEN_KEYFILES_DIRECTORY
+    ssh_pkey_dir.mkdir(parents=True, exist_ok=True)
+    private_keyfile_path = tempfile.NamedTemporaryFile(
+        prefix=f"{GEN_KEYFILE_PREFIX}_{credential.get('id')}_", dir=ssh_pkey_dir
+    ).name
+    with open(private_keyfile_path, "w+") as private_key_fp:
+        private_key_fp.write(ssh_keyvalue)
+        private_key_fp.write("\n")
+    os.chmod(private_keyfile_path, 0o600)
+    return private_keyfile_path
+
+
+def delete_ssh_keyfiles(inventory):
+    """Delete all generated ansible_ssh_private_key_files."""
+    if not inventory:
+        return
+
+    # Defined during connects
+    ansible_vars = inventory.get("all", {}).get("vars", {})
+    ssh_pkf = ansible_vars.get("ansible_ssh_private_key_file")
+    if ssh_pkf:
+        delete_ssh_keyfile(ssh_pkf)
+    # Defined during inspects
+    children = inventory.get("all", {}).get("children", {})
+    for group in children.keys():
+        hosts = children.get(group).get("hosts", {})
+        for host in hosts.keys():
+            ssh_pkf = hosts.get(host).get("ansible_ssh_private_key_file")
+            if ssh_pkf:
+                delete_ssh_keyfile(ssh_pkf)
+
+
+def is_gen_ssh_keyfile(private_keyfile_path):
+    """Return True if the file path specified is a generated ssh keyfile."""
+    keyfile_pattern = (
+        f"^{tempfile.gettempdir()}"
+        f"/{GEN_KEYFILES_DIRECTORY}/{GEN_KEYFILE_PREFIX}_[0-9]+_.*$"
+    )
+    return True if re.match(keyfile_pattern, private_keyfile_path) else False
```

Comment:
It's a real check,  whether we specify an ssh_keyfile, or the actual ssh_keyvalue, both methods use the one and only Ansible mechanism that is file driven and driven by the `ansible_ssh_private_key_file` parameter.  We need to make sure we don't zap the user-provided ssh_keyfiles.  Otherwise, the code would have been a bit simpler.

Reply:
Acknowledged!